cmd/ifupdown.c

```diff
@@ -215,8 +215,10 @@ change_auto_interfaces(struct lif_dict *collection, struct lif_dict *state, stru
 		    fnmatch(opts->include_pattern, iface->ifname, 0))
 			continue;
 
-		if (!change_interface(iface, collection, state, iface->ifname, false))
-			return false;
+		if (!change_interface(iface, collection, state, iface->ifname, false)) {
+			fprintf(stderr, "%s: failed to change state for interface %s\n", argv0, iface->ifname);
+			continue;
```

Comment:
With that change we always return true, so I guess we should remove the `bool` return value here, as it got redundant, and update the calling code.

@kaniini what do you think?

Reply:
Alternatively, we could also return false when one interfaces was not brought up successfully.

That is, change this to something along the lines of:

```C
bool ret = true;

LIF_DICT_FOREACH(iter, collection) {
  // …

  if (!change_interface(iface, collection, state, iface->ifname, false)) {
    ret = false;
    fprintf(…);
    continue;
  }
}

return ret;
```

Reply:
I think I would prefer to do what @nmeum wants to do here.

Reply:
Updated accordingly, please take another look.